app/models/package_manager/go.rb

```diff
@@ -21,32 +23,71 @@ class Go < Base
 
 
     def self.package_link(project, version = nil)
-      "http://go-search.org/view?id=#{project.name}"
+      "https://pkg.go.dev/#{project.name}#{"@#{version}" if version}"
     end
 
     def self.documentation_url(name, version = nil)
-      "http://godoc.org/#{name}"
+      "https://pkg.go.dev/#{name}#{"@#{version}" if version}?tab=doc"
     end
 
     def self.install_instructions(project, version = nil)
       "go get #{project.name}"
     end
 
     def self.project_names
-      get("http://go-search.org/api?action=packages")
+      # Currently the index only shows the last <=2000 modules from the date given. (https://proxy.golang.org/)
+      project_window = 1.day.ago.strftime("%FT%TZ")
+      get_raw("https://index.golang.org/index?since=#{project_window}&limit=2000")
+        .lines
+        .map { |line| JSON.parse(line)["Path"] }
     end
 
     def self.project(name)
-      get("http://go-search.org/api?action=package&id=#{name}")
+      if pkg_html = get_html("https://pkg.go.dev/#{name}?tab=doc")
+        go_module, _latest_version = pkg_html.css('a[data-test-id="DetailsHeader-infoLabelModule"]').first&.text&.split('@', 2)
```

Comment:
This seems fragile... how often is `name != go_module`? Because then we could just use the version list from the proxy and figure out latest version ourselves

Reply:
We're currently indexing both packages+modules, so `name` could be either the package or the parent module.

This part's not to get the latest_version tho, it's just to strip the latest_version from this module name. I might be wrong and it's not there, so I'll circle back in a bit and see if the splitting is necessary.